games/TicTacToe/src/utils.sw

```diff
@@ -0,0 +1,68 @@
+library utils;
+
+use core::ops::Eq;
+
+impl<T> Eq for Option<T> {
+    fn eq(self, other: Self) -> bool {
+        match (self, other) {
+            (Option::None, Option::None) => true,
+            (Option::Some(T), Option::Some(T)) => true,
+            _ => false,
+        }
+    }
+}
+
+pub fn win_check(board: Vec<Option<Identity>>, player: Option<Identity>) -> bool {
+    let matches = [
+        [0, 1, 2],
+        [3, 4, 5],
+        [6, 7, 8],
+        [0, 3, 6],
+        [1, 4, 7],
+        [2, 5, 8],
+        [0, 4, 8],
+        [2, 4, 6],
+    ];
+    let mut i = 0;
+    while (i < 9) {
+        if (board.get(matches[i][0]).unwrap() == player)
+            && (board.get(matches[i][1]).unwrap() == player)
+            && (board.get(matches[i][2]).unwrap() == player)
+        {
+            return true;
+        }
+        i += 1;
+    }
+    return false;
+}
+
+pub fn draw(
+    board: Vec<Option<Identity>>,
+    player_one: Option<Identity>,
+    player_two: Option<Identity>,
+) -> bool {
+    let matches = [
```

Comment:
I'm wondering if it would be more efficient to put the repeated matches variable in storage.  Do you know @simonr0204 ?

Reply:
hmm so since this is a constant, it's probably better for it to be in the bytecode rather than in storage. Storage reads are $$$